daemon/images/image_pull.go

```diff
@@ -44,7 +46,9 @@ func (i *ImageService) PullImage(ctx context.Context, image, tag, os string, met
 		}
 	}
 
-	return i.pullImageWithReference(ctx, ref, os, metaHeaders, authConfig, outStream)
+	err = i.pullImageWithReference(ctx, ref, os, metaHeaders, authConfig, outStream)
+	imageActions.WithValues("pull").UpdateSince(start)
```

Comment:
why isn't this just a defer?

Reply:
Because a defer wasn't necessary. If the function exits before this point then a pull didn't actually happen. Recording a time for a validation failure would produce misleading metrics.